src/datanode/src/instance.rs

```diff
@@ -409,14 +409,21 @@ fn create_object_store_with_cache(
     };
 
     if let Some(path) = cache_path {
-        let cache_store =
-            FsBuilder::default()
-                .root(path)
-                .build()
-                .with_context(|_| error::InitBackendSnafu {
-                    config: store_config.clone(),
-                })?;
-        let cache_layer = LruCacheLayer::new(Arc::new(cache_store), cache_capacity.0 as usize);
+        let atomic_temp_dir = format!("{path}/.tmp/");
+        clean_temp_dir(&atomic_temp_dir)?;
+        let cache_store = FsBuilder::default()
+            .root(path)
+            .atomic_write_dir(&atomic_temp_dir)
+            .build()
+            .with_context(|_| error::InitBackendSnafu {
+                config: store_config.clone(),
```

Comment:
question(out-of-topic): Is it necessary to put the config in the InitBackend error? The struct is quite large and we already have the error message from the source error.

Reply:
Agree, let me remove it.